src/integrationtests/java/com/aws/greengrass/integrationtests/metrics/MetricsEmitterTest.java

```diff
@@ -0,0 +1,207 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+public class MetricsEmitterTest {
+    private static final String MOCK_THING_NAME = "mockThing";
+    private static final long TEST_TIME_OUT_SEC = 10L;
+    private static final long AGGREGATE_INTERVAL = 2L;
+    private static final int DEFAULT_PUBLISH_INTERVAL = 86_400;
+    private Optional<MockedStatic<Clock>> clockMock;
+    private Kernel kernel;
+    private DomainEvents domainEvents;
+    @Mock
+    private MqttClient mqttClient;
+    @TempDir
+    Path rootDir;
+
+    @BeforeEach
+    void beforeEach() {
+        clockMock = Optional.empty();
+        domainEvents = new DomainEvents();
+        kernel = new Kernel();
+        kernel.getContext().put(DomainEvents.class, domainEvents);
+        kernel.getContext().get(CertificateSubscriptionEventHandler.class).listen();
+        kernel.getContext().get(VerifyClientDeviceIdentityEventHandler.class).listen();
+        kernel.getContext().get(AuthorizeClientDeviceActionsMetricHandler.class).listen();
+        kernel.getContext().get(SessionCreationEventHandler.class).listen();
+        kernel.getContext().get(ServiceErrorEventHandler.class).listen();
+        TelemetryConfig.getInstance().telemetryLoggerNamesSet.clear();
+        lenient().when(mqttClient.publish(any())).thenReturn(CompletableFuture.completedFuture(0));
+    }
+
+    @AfterEach
+    void afterEach() {
+        this.clockMock.ifPresent(ScopedMock::close);
+        if (kernel != null) {
+            kernel.shutdown();
+        }
+    }
+
+    private void startNucleusWithConfig(String configFileName) throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFileName).toString());
+        kernel.getContext().addGlobalStateChangeListener((service, was, newState) -> {
+            if (ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME.equals(service.getName())
+                    && service.getState().equals(State.RUNNING)) {
+                authServiceRunning.countDown();
+            }
+        });
+        kernel.launch();
+        assertThat(authServiceRunning.await(TEST_TIME_OUT_SEC, TimeUnit.SECONDS), is(true));
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
+    private void mockInstant(long expected) {
+        this.clockMock.ifPresent(ScopedMock::close);
+        Clock spyClock = spy(Clock.class);
+        MockedStatic<Clock> clockMock;
+        clockMock = mockStatic(Clock.class);
+        clockMock.when(Clock::systemUTC).thenReturn(spyClock);
```

Comment:
why are you mocking a static?

Clock should be injected via Context so that it is very easy to replace if you want to replace it.

Use `kernel.getContext().put(mockClock, Clock.class)` before calling `kernel.launch`.